src/Bindings/ManualBindings.cpp

```diff
@@ -3116,30 +3114,57 @@ static int tolua_cLineBlockTracer_Trace(lua_State * tolua_S)
 	cLuaState L(tolua_S);
 	if (
 		!L.CheckParamUserType(idx, "cWorld") ||
-		!L.CheckParamTable   (idx + 1) ||
-		!L.CheckParamNumber  (idx + 2, idx + 7) ||
-		!L.CheckParamEnd     (idx + 8)
+		!L.CheckParamTable   (idx + 1)
 	)
 	{
 		return 0;
 	}
 
 	// Get the params:
 	cWorld * world;
-	double startX, startY, startZ;
-	double endX, endY, endZ;
+	Vector3d start;
+	Vector3d end;
 	cLuaState::cTableRefPtr callbacks;
-	if (!L.GetStackValues(idx, world, callbacks, startX, startY, startZ, endX, endY, endZ))
+	if (
+		L.CheckParamNumber  (idx + 2, idx + 7) &&
+		L.CheckParamEnd     (idx + 8)
+	)
+	{
+		if (!L.GetStackValues(idx, world, callbacks, start.x, start.y, start.z, end.x, end.y, end.z))
+		{
+			LOGWARNING("cLineBlockTracer:Trace(): Cannot read parameters (starting at idx %d), aborting the trace.", idx);
+			L.LogStackTrace();
+			L.LogStackValues("Values on the stack");
+			return 0;
+		}
+		else
+		{
+			LOGWARNING("cLineBlockTracer:Trace(): Deprecated overload should take two Vector3d.");
+		}
+
+	}
+	else if (
+		L.CheckParamNumber  (idx + 2, idx + 3) &&
+		L.CheckParamEnd     (idx + 4) &&
+		L.CheckParamUserType(idx + 2, "Vector3d") &&
+		L.CheckParamUserType(idx + 3, "Vector3d")
+	)
+	{
+		if (!L.GetStackValues(idx, world, callbacks, start, end))
+		{
+			LOGWARNING("cLineBlockTracer:Trace(): Cannot read parameters (starting at idx %d), aborting the trace.", idx);
+			L.LogStackTrace();
+			L.LogStackValues("Values on the stack");
+			return 0;
+		}
+	}
+	else
 	{
-		LOGWARNING("cLineBlockTracer:Trace(): Cannot read parameters (starting at idx %d), aborting the trace.", idx);
-		L.LogStackTrace();
-		L.LogStackValues("Values on the stack");
```

Comment:
Should the diagnostic be left in here?

Also as an side, what's the "Values on the stack" message, it looks like debugging code left in...

Reply:
It was allready present for the other overload so I left it.

Reply:
The `Values on the stack` lists the Lua parameters, so the plugin author can see what got passed to the API function (-> why it was rejected). That should stay.